RecoPixelVertexing/PixelTrackFitting/plugins/PixelTrackFilterByKinematicsProducer.cc

```diff
@@ -43,7 +43,7 @@ void PixelTrackFilterByKinematicsProducer::fillDescriptions(edm::ConfigurationDe
 
   desc.add<double>("ptMin", 0.1);
   desc.add<double>("nSigmaInvPtTolerance", 0.0);
-  desc.add<double>("tipMax", 1.0);
+  desc.add<double>("tipMax", 0.1);
```

Comment:
Is it intentional to change the default for everyone? If not, the customization would be better done in the configuration with a modifier.

Reply:
Yes, it is.
Do you mean that it would be better to have a different cut for each fit? I don't really agree, since the the resolutions on the tip are significantly smaller than tipMax (see https://uz.sns.it/~Gimmy/broken_line/plots/plots_pixelTrack_pixel/resolutionsEta.pdf). I mean, the tip is a characteristic of a track, and it depends on the fit used only at a "higher order". So if a track is accepted/rejected for a fit, it will very likely accepted/rejected for another fit. And indeed the change in efficiency and fake rate was extremely similar for broken line, reference and Riemann fit.

Reply:
Since I'm not a tracking expert,  which of the plots shows it ?

Reply:
Clearly, the plots Efficiency vs tipMax and FakeRate vs tipMax do not exist, since they would need a run for each point in the plot.
The value of 1mm was chosen by looking at the "suggestion" given by the fakerate vs dxy(PV) plot (fourth plot here https://uz.sns.it/~Gimmy/broken_line/plots/plots_pixelTrack_pixel/dupandfakeDxyDzPV.pdf). The most useful plot for the tuning would be the first here https://uz.sns.it/~Gimmy/broken_line/plots/plots_pixelTrack_pixel/dupandfakeDxyDzBS.pdf but there are evidently problems with the scale of the x axis. I tried to change the scale and the number of bins with Felice, but with little success.

However, when I was making the "final" plots for the report I'm writing (they can be found at https://uz.sns.it/~Gimmy/broken_line/plots/), which are made with tipMax=0.1 for all the three fits, I initially forgot to change tipMax from 1 to 0.1 in the release in which I was running the reference. Fortunately, I saved these plots (https://uz.sns.it/~Gimmy/plots/), so you can compare the reference performance with tipMax=1 and with tipMax=0.1. For example, looking at the Summary, you can see that the efficiency drops from 0.4997 to 0.4949 (-0.96%), while the fake rate drops from 0.0529 to 0.0207 (-60.87%).

Reply:
@GimmyTomas thank you for the detailed comparison.
Looking at the two summaries:

  | BrokenLine | RiemannFit | Reference
-- | -- | -- | --
Efficiency | 0.4948 | 0.4949 | 0.4997
Number of TrackingParticles (after cuts) | 362210 | 362210 | 362210
Number of matched TrackingParticles | 179209 | 179252 | 180986
Fake rate | 0.0232 | 0.0229 | 0.0529
Duplicate rate | 0.0489 | 0.0482 | 0.0448
Number of tracks | 2154268 | 2152740 | 2236461
Number of true tracks | 2104359 | 2103498 | 2118150
Number of fake tracks | 49909 | 49242 | 118311
Number of pileup tracks | 1854368 | 1853567 | 1865766
Number of duplicate tracks | 105402 | 103767 | 100147

and


  | BrokenLine | RiemannFit | Reference
-- | -- | -- | --
Efficiency | 0.4948 | 0.4949 | 0.4949
Number of TrackingParticles (after cuts) | 362210 | 362210 | 362210
Number of matched TrackingParticles | 179209 | 179252 | 179262
Fake rate | 0.0232 | 0.0229 | 0.0207
Duplicate rate | 0.0489 | 0.0482 | 0.0462
Number of tracks | 2154268 | 2152740 | 2134656
Number of true tracks | 2104359 | 2103498 | 2090443
Number of fake tracks | 49909 | 49242 | 44213
Number of pileup tracks | 1854368 | 1853567 | 1841564
Number of duplicate tracks | 105402 | 103767 | 98546

I take it that only the **Reference** column has 1.0 vs 0.1, while the **BrokenLine** and **RiemannFit** columns always have 0.1 .

Do you still have the original root files ?
It would be interesting to see one comparison with
  - reference with 1.0
  - reference with 0.1
  - Riemann fit with 0.1
  - broken line fit with 0.1

That said,  I would prefer not to make the global change from `1.0` to `0.1` as part of this PR, but in a separate one (if the tracking experts agree, of course).

Reply:
Exactly: as I said, I initially just forgot to change from 1.0 to 0.1 in the release in which I was running the reference, so it is the only fit with this change, the others always have the 0.1 cut.
If I'm not wrong, I have the original root files only for the broken line and the Riemann fit (so I have neither the reference with 1.0, neither the reference with 0.1), unfortunately.

Reply:
I was mostly concerned by that in principle there can be other applications in CMSSW using the pixel tracks, and we shouldn't just "blindly" set the default for everyone. Ok, in practice I only see some online DQM clients using pixel tracks, and we're in our own repository/branch, so maybe it doesn't matter. If this goes in, #160 has be to adapted to change it back to 1.0 (unless HLT changes it as well).

I'd also prefer to do this change in a separate PR.